backend/src/youtube_handler.py

```diff
@@ -0,0 +1,136 @@
+    api_version = "v3"
+    MAX_THREAD_COUNT: int = 5
+    _youtube_instance: discovery.Resource | None = None
+    request_queue: queue.Queue = queue.Queue()
+    thread_pool: list[threading.Thread] = []
+
+
+    def __init__():
+        """
+        Raises a TypeError.
+        YoutubeHandler instances must not be created as this is a singleton
+        """
+        raise TypeError(
+            "YoutubeHandler instances should not be created. " +
+            "Consider using the `get_client()` function."
+        )
+    @staticmethod
+    def get_client():
+        """
+        Retrieves or creates the YouTube client instance
+
+        Returns
+        -------
+        googleapiclient.discovery.Resource
+            The static YouTube client
+        """
+        if YoutubeHandler._youtube_instance:
+            return YoutubeHandler._youtube_instance
+
+        return YoutubeHandler._initialize_client()
+    @staticmethod
+    def _initialize_client():
+        """
+        Initializes the YouTube client.
+        """
+        _youtube_instance = discovery.build(
+            YoutubeHandler.api_service_name,
+            YoutubeHandler.api_version,
+            developerKey=SecretsHandler.get_youtube_key()
+        )
+        return _youtube_instance
+    
+    @staticmethod
+    def search_for_match(song: dict):
+        """
+        Searches for a matching song on youtube given the song and artist name
+        Adds the youtube url to the song dictionary
+        if no match is found, the song is returned as None
+
+        Returns
+        -------
+        googleapiclient.discovery.Resource
+            The search results
+        """
+        youtube_url = "https://www.youtube.com/watch?v=%s"
+        response = None
+        try:
+            response = YoutubeHandler.get_client().search().list(
+                part = "id",
+                q = song["name"] + " " + song["artists"][0]["name"],
+                type = "video",
+                maxResults = 1,
+                order = "relevance",
+                fields = "items(id(videoId))"
+            ).execute()
+        except HttpError as e:
+            # If the request fails, it probably means the quota has been exceeded
+            # If this happens too often, consider applying for a quota increase at https://support.google.com/youtube/contact/yt_api_form
+            # For now, we'll just ignore the error and return nothing
+            pass
+        if response and response["items"]:
+            song["youtube_url"] = youtube_url % response["items"][0]["id"]["videoId"]
+        else:
+            song["youtube_url"] = ""
+        return song 
+    
+    @classmethod
+    def match_list(cls, songs: list[dict[str, str | bool | int]]):
+        """
+        Matches a list of songs to their respective youtube videos
+
+        Parameters
+        ----------
+        songs : list[dict[str, str | bool | int]]
+            A list of songs to match
+        """
+        queue
```

Comment:
I'm not sure if GitHub is glitching, but this looks like just a line with the word "queue" on it?

Reply:
idek where that came from and I'm surprised the file ran, will be deleted